Source/WebCore/rendering/svg/SVGTextLayoutEngine.cpp

```diff
@@ -539,7 +539,7 @@ void SVGTextLayoutEngine::layoutTextOnLineOrPath(InlineIterator::SVGTextBoxItera
         updateRelativePositionAdjustmentsIfNeeded(data.dx, data.dy);
 
         // Calculate CSS 'letter-spacing' and 'word-spacing' for next character, if needed.
-        float spacing = spacingLayout.calculateCSSSpacing(currentCharacter);
+        float spacing = spacingLayout.calculateCSSSpacing(currentCharacter ? std::optional<char16_t>(*currentCharacter) : std::nullopt);
```

Comment:
Why do we have to use `std::optional` or raw pointer to pass the currentCharacter? Can we just pass zero if `currentCharacter` is null?
```
float spacing = spacingLayout.calculateCSSSpacing(currentCharacter ? *currentCharacter : '\0');
```

Reply:
I did pass null earlier but then switched to std::optional. I can go back as well.